### PR TITLE
ci: add chromatic

### PR DESCRIPTION
**Pull request by benelan**
**Related Issue:** https://github.com/Esri/calcite-components/issues/4825

## Summary

Add chromatic to our CI

### CODE REVIEW

File: src/components/progress/progress.stories.ts [hunk 1]
```diff
@@ -30,4 +30,4 @@ export const darkThemeRTL_TestOnly = (): string => html`
   ></calcite-progress>
 `;
 
-darkThemeRTL_TestOnly.parameters = { themes: themesDarkDefault };
+darkThemeRTL_TestOnly.parameters = { themes: themesDarkDefault, chromatic: { disableSnapshot: true } };
```
Comment (jcfranco) [inline src/components/progress/progress.stories.ts:33]: Based on the above changes, is it intentional for this to have both the `_TestOnly` suffix and disabled snapshot?

File: .storybook/main.ts [hunk 2]
```diff
@@ -5,7 +5,9 @@ module.exports = {
     "@storybook/addon-a11y",
     "@whitespace/storybook-addon-html",
     "storybook-rtl-addon",
-    "storybook-addon-themes"
+    "storybook-addon-themes",
```
Comment (jcfranco) [inline .storybook/main.ts:8]: Nitpick: can these be sorted alphabetically?

File: .storybook/helpers.ts [hunk 3]
```diff
@@ -11,7 +11,7 @@ export const iconNames = Object.keys(icons)
   .map((iconName) => iconName.replace("16", ""));
 
 // custom boolean will start up a knob but only add the prop if it is true
-// if you'd insead like `attr="true|false" set the standalone option to false
+// if you'd instead like `attr="true|false" set the standalone option to false
```
Comment (jcfranco) [inline .storybook/helpers.ts:14]: 🏆

**Review by jcfranco — APPROVED** [review]
giphy

🤫💰

File: src/05-custom-theme.stories.mdx [hunk 4]
```diff
@@ -182,7 +182,7 @@ export const Template = () =>
             <calcite-tab-title>Tab 4 Title</calcite-tab-title>
           </calcite-tab-nav>
         </calcite-tabs>
-        <calcite-loader active></calcite-loader>
+        <calcite-loader class="chromatic-ignore" active></calcite-loader>
```
Comment (driskull) [inline src/05-custom-theme.stories.mdx:185]: soooo coool

File: src/components/progress/progress.stories.ts (hunk 1 above)
Comment (benelan) [reply src/components/progress/progress.stories.ts:33]: Only for now while we are running Screener and Chromatic side by side. I'll clean it up once we completely remove screener

**Review by eriklharper — APPROVED** [review]

**Pull request merged**